libs/permissions-client/src/permissions_client/core.clj

```diff
@@ -0,0 +1,202 @@
+     resources may not be deleted.")
+
+  (update-resource-type [_ id resource-type-name description]
+    "Updates a resource type in the permissions service. Each resource type must have a unique name.")
+
+  (list-permissions [_]
+    "Lists all permissions known to the permissions service.")
+
+  (grant-permission [_ resource-type resource-name subject-type subject-id level]
+    "Grants permission to access a resource to a user. The resource-type, resource-name, subject-type, and
+     subject-id fields have the same meanings as in the resources and subjects methods. Neither the resource
+     nor the subject need to be registered before calling this ednpoint; they will be added to the database
+     if necessary. The permission level must correspond to one of the available levels in the permissions
+     service. The currently available levels are 'read', 'admin', 'write', and 'own'.")
+
+  (revoke-permission [_ resource-type resource-name subject-type subject-id]
+    "Revokes a permission that has previously been granted.")
+
+  (list-resource-permissions [_ resource-type resource-name]
+    "Lists all permissions associated with a resource.")
+
+  (get-subject-permissions [_ subject-type subject-id lookup?]
+    "Looks up permissions that have been granted to a subject. If the 'lookup?' flag is set to 'true' and the
+     subject happens to be a user then the most privileged permissions available to the user or any group that
+     the user belongs to (as determined by Grouper) will be listed. If the 'lookup?' flag is set to 'false' or
+     the subject is a group then only permissions that were granted directly to the subject will be listed.")
+
+  (get-subject-permissions-for-resource-type [_ subject-type subject-id resource-type lookup?]
+    "Looks up permissions that have been granted to a subject for a single resource type. If the 'lookup?' flag
+     is set to 'true' and the subject happens to be a user then the most privileged permissions available to the
+     user or any group that the user belongs to (as determined by Grouper) will be listed. If the 'lookup?'
+     flag is set to 'false' or the subject is a group then only permissions that were granted directly to the
+     subject will be listed.")
+
+  (get-subject-permissions-for-resource [_ subject-type subject-id resource-type resource-name lookup?]
+    "Looks up permissions that have been granted to a subject for a single resource. If the 'lookup?' flag is
+     set to 'true' and the subject happens to be a user then the most privileged permissions available to the
+     user or any group that the user belongs to (as determined by Grouper) will be listed. If the 'lookup?'
+     flag is set to 'false' or the subject is a group then only permissions that were granted directly to the
+     subject will be listed."))
+
+(defn- build-url [base-url & path-elements]
+  (str (apply curl/url base-url path-elements)))
+
+(deftype PermissionsClient [base-url]
+  Client
```

Comment:
Very clean way of allowing the `base-uri` to be configurable outside of this library 👍